components/scream/src/physics/p3/p3_functions_incloud_mixingratios_impl.hpp

```diff
@@ -0,0 +1,91 @@
+#ifndef P3_FUNCTIONS_INCLOUD_MIXINGRATIOS_IMPL_HPP
+#define P3_FUNCTIONS_INCLOUD_MIXINGRATIOS_IMPL_HPP
+
+#include "p3_functions.hpp" // for ETI only but harmless for GPU
+#include "p3_functions_math_impl.hpp"
+
+namespace scream {
+namespace p3 {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::calculate_incloud_mixingratios(const Spack& qc, const Spack& qr, const Spack& qitot, const Spack& qirim, const Spack& nc,
+                                 const Spack& nr, const Spack& nitot, const Spack& birim, const Spack& inv_lcldm, 
+                                 const Spack& inv_icldm, const Spack& inv_rcldm, 
+                                 Spack& qc_incld, Spack& qr_incld, Spack& qitot_incld, Spack& qirim_incld,
+                                 Spack& nc_incld, Spack& nr_incld, Spack& nitot_incld, Spack& birim_incld)
+{
+   constexpr Scalar qsmall = C::QSMALL;
+   constexpr Scalar incloud_limit = C::incloud_limit;
+   constexpr Scalar precip_limit  = C::precip_limit;
+
+   const auto qc_ge_qsmall = qc >= qsmall;
+   const auto not_qc_ge_qsmall = ! qc_ge_qsmall;
+
+   const auto qitot_ge_qsmall = qitot >= qsmall;
+   const auto not_qitot_ge_qsmall = ! qitot_ge_qsmall;
+
+   const auto qitot_and_qirim_ge_qsmall = (qitot >= qsmall) && (qirim >= qsmall);
+   const auto not_qitot_and_qirim_ge_qsmall = ! qitot_and_qirim_ge_qsmall;
+
+   const auto qr_ge_qsmall = qr >= qsmall;
+   const auto not_qr_ge_qsmall = ! qr_ge_qsmall;
+
+   qc_incld.set(qc_ge_qsmall,
+                qc*inv_lcldm);
+   nc_incld.set(qc_ge_qsmall,
+                pack::max(nc*inv_lcldm,sp(0)));
```

Comment:
You don't need `sp(0)`, just `0` should be fine. Make this change here and all the places it occurs below.